### PR TITLE
[WFCORE-5343]: Preparing YAML Support

### PR DESCRIPTION
**Pull request by ehsavoie**
* Adding metadata for the underlying types for a list attrbite definition.

Thanks for submitting your Pull Request!

Please delete this text, and add a link to the Jira issue solved by this PR.

Remember to use the Jira issue ID in the PR title, and any commits.

### CODE REVIEW

**Review by bstansberry — CHANGES_REQUESTED** [review]
As discussed at https://github.com/wildfly/wildfly/pull/15006#issuecomment-988123885 the TimerAttributeDefinition and LegacySupport changes should not be needed.

**Comment by ehsavoie** [comment]
Fixed and rebased

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/AppClientMain.java [hunk 1]
```diff
@@ -46,7 +46,6 @@ public static void main(final String[] params) {
             logger.error("InAppClientContainer was not true");
             throw new RuntimeException("InAppClientContainer was not true");
         }
-
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ee/appclient/basic/AppClientMain.java:49]: Please restore this line (just to get this commit out of the file's history.)

**Review by bstansberry — CHANGES_REQUESTED** [review]
@ehsavoie The test aspects of this seem to be WIP that needs to be completed.

**Comment by ehsavoie** [comment]
@bstansberry all should be fixed now :)

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @ehsavoie

**Pull request merged**